unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs

```diff
@@ -166,18 +198,31 @@ internal IEnumerator InstantiateItemOnEditor(BaseComponentView newItem, string n
         InstantiateItem(newItem, name);
     }
 
-    internal void RemoveAllInstantiatedItems()
+    internal void DestroyInstantiatedItems(bool forzeToDestroyAll)
     {
-        foreach (Transform child in transform)
+        if (forzeToDestroyAll)
         {
-            if (Application.isPlaying)
+            foreach (Transform child in transform)
             {
-                Destroy(child.gameObject);
+                if (Application.isPlaying)
+                {
+                    Destroy(child.gameObject);
+                }
+                else
+                {
+                    if (isActiveAndEnabled)
+                        StartCoroutine(DestroyGameObjectOnEditor(child.gameObject));
+                }
```

Comment:
This is quite dangerous. 

If we want to diverge the code from builds to editor let's use preprocessing directives:
```
#if UNITY_EDITOR
  Destroy(child.gameObject);
#else
  StartCoroutine(DestroyGameObjectOnEditor(child.gameObject));
#endif
```

Reply:
Here we don't want to separate the build behaviour from the editor one but the inspector behaviour (for the cases when we are configuring an UI component from its inspector and it needs to be refreshed after a frame) from the play mode (where the component refreshing happens normally).

Reply:
Aaahhh, I would still wrap the part that is only relevant for the Editor in a preprocessing directive. `DestroyImmediate` leads to serious issues and better be safe than sorry and ensuring there's no way we can reach that code in production.